Base/Implementations/PropertyRule.cs

```diff
@@ -57,6 +63,7 @@ public void Update()
                 {
                     if (!r.Expression())
                     {
+                        _errorMessages.Add(new ErrorMessage(r.Message, r.ErrorLevel));
```

Comment:
could we just add on line 67 _errors.Add(string.Format("Error Level: {0} Error: {1}\n", r.ErrorLevel, r.Message));

Reply:
But then it is still just a string. When you ReportMessage how does the message hub know how to filter it based on its error level. That was the problem I was trying to solve.

Reply:
ah. so what about having the method and dynamically creating the list  as i comment on line 99 of validation.cs

Reply:
we can always refactor and optimize if there is a memory issue with the changes...

Reply:
I like that. I'm on it.